internal/eva/client.go

```diff
@@ -0,0 +1,26 @@
+package eva
+
+import (
+	"github.com/go-resty/resty/v2"
+)
+
+const (
+	contentType = "application/json"
+	userAgent   = "terraform-provider-eva"
+)
+
+type Client struct {
+	Client *resty.Client
```

Comment:
Unless we wanna enable easy debugging this should be a private field, since the resty client exposes low-level access that we don't otherwise wanna give to higher levels in the stack.

Reply:
Further, to avoid confusion it would be nice to rename this to either `restClient` or `httpClient`. Otherwise without the context a call to `client.Client` becomes hard to read.

Reply:
Need to expose the rest client, because the `login` function appends the `authorization` header to the client which would then be used for all other requests afterwards.

Reply:
Fixed it be exposing a `setAuthorizationToken` method on the `Client`.